extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -138,13 +139,17 @@ public MSQCompactionRunner(
    */
   @Override
   public CompactionConfigValidationResult validateCompactionTask(
-      CompactionTask compactionTask
+      CompactionTask compactionTask,
+      Map<Interval, DataSchema> intervalToDataSchemaMap
   )
   {
     List<CompactionConfigValidationResult> validationResults = new ArrayList<>();
     if (compactionTask.getTuningConfig() != null) {
-      validationResults.add(ClientCompactionRunnerInfo.validatePartitionsSpecForMSQ(
-          compactionTask.getTuningConfig().getPartitionsSpec())
+      validationResults.add(
+          ClientCompactionRunnerInfo.validatePartitionsSpecForMSQ(
+              compactionTask.getTuningConfig().getPartitionsSpec(),
+              Iterables.getOnlyElement(intervalToDataSchemaMap.values()).getDimensionsSpec().getDimensions()
```

Comment:
Was meaning to add a check for that since we are currently not able to handle multiple intervals in the map for multiple reasons, one of them being that the subsequent worker ids clash -- since they are derived from controller task ID which in turn is equal to Compaction task ID each time (they are all launched sequentially in-place inside the MSQCompactionRunner).

Have added the check now.

Reply:
Okay. Could you please add a comment for the reason of this expectation?